app/uk/gov/hmrc/exports/controllers/HeaderValidator.scala

```diff
@@ -82,14 +82,30 @@ class HeaderValidator {
     }
   }
 
-  def validateAndExtractNotificationHeaders(
+  def validateAndExtractMovementNotificationHeaders(
     headers: Map[String, String]
-  ): Either[ErrorResponse, ValidatedHeadersNotificationApiRequest] = {
+  ): Either[ErrorResponse, MovementNotificationApiRequest] = {
     val result = for {
       eori <- extractEoriHeader(headers)
       authToken <- extractAuthTokenHeader(headers)
       conversationId <- extractConversationIdHeader(headers)
-    } yield ValidatedHeadersNotificationApiRequest(AuthToken(authToken), ConversationId(conversationId), Eori(eori))
+    } yield MovementNotificationApiRequest(AuthToken(authToken), ConversationId(conversationId), Eori(eori))
+    result match {
+      case Some(vhnar) => Right(vhnar)
+      case _ => {
+        Logger.debug("Error validating and extracting headers")
+        Left(ErrorResponse.ErrorInvalidPayload)
+      }
+    }
+  }
+
+  def validateAndExtractSubmissionNotificationHeaders(
+    headers: Map[String, String]
+  ): Either[ErrorResponse, SubmissionNotificationApiRequest] = {
+    val result = for {
+      authToken <- extractAuthTokenHeader(headers)
+      conversationId <- extractConversationIdHeader(headers)
+    } yield SubmissionNotificationApiRequest(AuthToken(authToken), ConversationId(conversationId))
     result match {
       case Some(vhnar) => Right(vhnar)
```

Comment:
Here probably `vhnar` should be changed too

Reply:
named them all request for ease